src/librustc_hir/hir.rs

```diff
@@ -1330,6 +1330,18 @@ pub struct Expr<'hir> {
     pub span: Span,
 }
 
+/// `Expr` pattern alias to facilitate pattern-matching.
+///
+/// Usage examples:
+/// * `hir::Expr!(Unary(_, hir::Expr!(Lit(_))))`
+/// * `hir::Expr! { Loop(..), hir_id: loop_id }`
+pub macro Expr($variant:ident $($rest:tt)*) {
+    $crate::hir::Expr {
+        kind: $crate::hir::ExprKind::$variant $($rest)*,
```

Comment:
`$crate` -> `crate` here and one line above, `macro`s are fully hygienic and don't need `$crate`.

Reply:
I wanted to be conservative, presumably I can even write `Expr` without the full module path.